python/coinbase-agentkit/coinbase_agentkit/action_providers/erc20/erc20_action_provider.py

```diff
@@ -0,0 +1,124 @@
+"""ERC20 action provider implementation."""
+
+from web3 import Web3
+
+from ...network import Network
+from ...wallet_providers import EvmWalletProvider
+from ..action_decorator import create_action
+from ..action_provider import ActionProvider
+from .constants import ERC20_ABI
+from .schemas import GetBalanceSchema, TransferSchema
+
+
+class ERC20ActionProvider(ActionProvider):
```

Comment:
```suggestion
class ERC20ActionProvider(ActionProvider[EvmWalletProvider]):
```
remember to add the wallet provider type arg

Reply:
omg, i know i put that in there at one point...arrrr